akka-typed/src/main/scala/akka/typed/ScalaProcess.scala

```diff
@@ -435,10 +436,16 @@ object ScalaProcess {
    * target when the watched actor has terminated. The returned Cancellable
    * can be used to unwatch the watchee, which will inhibit the message from
    * being dispatched—it might still be delivered if it was previously dispatched.
+   *
+   * If `onFailure` is provided it can override the value to be sent if the
+   * watched Actor failed and was a child Actor of the Actor hosting this process.
```

Comment:
sounds more like `mapWatchMessage` or something like that? We have mapError in streams so being similar would be nice